### DISCUSSION TITLE
Decentralizing farcaster channels

### DISCUSSION THREAD

**Discussion by thibauld** [farcasterxyz/protocol#150] [FIP Stage 1: Ideas]
I wanted to post here a proposal to decentralize Farcaster channels that I casted on Farcaster recently and that received very [warm](https://warpcast.com/vitalik.eth/0xfc2ff391) support: https://warpcast.com/~/conversations/0xf8ccabfbdcbd5528922ce84124ccd29e9f3b4490

TL;DR Channels identity should be separated from their name. Like for usernames, a channel identity should be an identifier (FCID - Farcaster Channel ID) and the association name <-> FCID should be done at the app layer. This will allow the protocol to not have to deal with squatting and impersonation issues. It will also allow for more innovation at the app layer, notably when it comes to deal with squatting and impersonation issues (outside of the scope of that discussion because it's not protocol level).

I suppose that you guys already had this approach in mind (as it's close to how usernames are being handled today) but, just in case, I thought I'd start the discussion. Also, I realize that this is just a general idea and far from being a specification at this stage.

**Comment by varunsrin** [comment]
Closing since this isn't a proposal.
